### PR TITLE
Fixes #32000 - Remove legacy export/import API endpoints

### PR DESCRIPTION
**Pull request by chris1984**
Part of 6.10 Pulp2 legacy clean up, will link the hammer PR when I am finished. 

Hammer PR: https://github.com/Katello/hammer-cli-katello/pull/788

### CODE REVIEW

File: app/controllers/katello/api/v2/content_view_versions_controller.rb [hunk 1]
```diff
@@ -85,49 +84,6 @@ def republish_repositories
...
-      end
-
-      if params[:since].present?
-        begin
-          params[:since].to_datetime
-        rescue
-          raise HttpErrors::BadRequest, _("Invalid date provided.")
-        end
-      end
-      task = async_task(::Actions::Katello::ContentViewVersion::Export, @content_view_version,
```
Comment (jturel) [inline app/controllers/katello/api/v2/content_view_versions_controller.rb:114]: Drive-by review! Are this action and the below ::Import still used?

Comment (chris1984) [reply app/controllers/katello/api/v2/content_view_versions_controller.rb:114]: Good catch, @parthaa do you know if we still use this?

Comment (jturel) [reply app/controllers/katello/api/v2/content_view_versions_controller.rb:114]: `git grep ContentViewVersion::Export` should give a pretty clear answer :)

Comment (chris1984) [reply app/controllers/katello/api/v2/content_view_versions_controller.rb:114]: Haha fair, ok ill unping Partha

Comment (chris1984) [reply app/controllers/katello/api/v2/content_view_versions_controller.rb:114]: Looks like all this stuff is still being used in places, for now will just close this and go with the hammer pr

**Comment by jturel** [comment]
Looks good at a glance. Will test it when I get some time. In the mean time you may wanna looks for some checks we also do like this which can also be cleaned up:

https://github.com/Katello/katello/blob/1cb62370d82cca17806420c0b1f671a95a65a71f/app/lib/actions/katello/content_view_version/import.rb#L7-L9

**Comment by jturel** [comment]
In content_views_controller there's a `pulp3_yum?` method that can be removed!

File: app/controllers/katello/api/v2/content_export_incrementals_controller.rb [hunk 2]
```diff
@@ -1,6 +1,5 @@
 module Katello
   class Api::V2::ContentExportIncrementalsController < Api::V2::ApiController
-    before_action :fail_if_not_pulp3, :only => [:version, :library]
```
Comment (jturel) [inline app/controllers/katello/api/v2/content_export_incrementals_controller.rb:3]: I see a lot of test stubs referring to `fail_if_not_pulp3`

**Review by jturel — CHANGES_REQUESTED** [review]

**Comment by chris1984** [comment]
@jturel I think I got them all this time, i feel stupid for missing the test stubs :(

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/content-view-details.controller.js [hunk 3]
```diff
@@ -36,8 +36,7 @@ angular.module('Bastion.content-views').controller('ContentViewDetailsController
             publish: "Actions::Katello::ContentView::Publish",
             promotion: "Actions::Katello::ContentView::Promote",
             removal: "Actions::Katello::ContentView::Remove",
-            incrementalUpdate: "Actions::Katello::ContentView::IncrementalUpdates",
-            export: "Actions::Katello::ContentViewVersion::Export"
+            incrementalUpdate: "Actions::Katello::ContentView::IncrementalUpdates"
```
Comment (jturel) [inline engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/content-view-details.controller.js:39]: PR LGTM but I have a question about the removal here. This code relates to the History tab on the CV details page which shows details about all of the tasks listed here, so this change means it won't show exports any more.

If it's a drop-in replacement, I think it's worth it to just change to the new task name `::Actions::Pulp3::Orchestration::ContentViewVersion::Export`

Alternatively, I see a good argument for keeping the Actions::Katello::ContentViewVersion::Export task and just make it plan `::Actions::Pulp3::Orchestration::ContentViewVersion::Export` and revert the change in this file. The main benefit being not mentioning "pulp3 orchestration" as a top-level task from Katello. 

Note there was also a BZ for it, so removing it could break QE automation: https://bugzilla.redhat.com/show_bug.cgi?id=1311526

Thoughts?

Comment (chris1984) [reply engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/content-view-details.controller.js:39]: I like the 2nd approach better, since it allows us to keep the name which won't break anything for QE/users and we can just under the hood have it plan the `::Actions::Pulp3::Orchestration::ContentViewVersion::Export` Will update and request a review when I am finished.

File: app/controllers/katello/api/v2/content_exports_controller.rb [hunk 4]
```diff
@@ -41,7 +34,7 @@ def api_status
     param :fail_on_missing_content, :bool, :desc => N_("Fails if any of the repositories belonging to this version"\
                                                          " are unexportable. False by default."), :required => false
     def version
-      tasks = async_task(::Actions::Pulp3::Orchestration::ContentViewVersion::Export,
+      tasks = async_task(::Actions::Katello::ContentViewVersion::Export,
```
Comment (jturel) [inline app/controllers/katello/api/v2/content_exports_controller.rb:37]: I think the incremental export controller needs this same change

File: app/lib/actions/katello/content_view_version/export.rb [hunk 5]
```diff
@@ -2,39 +2,8 @@ module Actions
...
-
-        def finalize
-          history = ::Katello::ContentViewHistory.find(input[:history_id])
-          history.status = ::Katello::ContentViewHistory::SUCCESSFUL
-          history.save!
-        end
-
-        def rescue_strategy
-          Dynflow::Action::Rescue::Skip
+        def plan(id, destination_server, chunk_size_mb, fail_on_missing_content)
```
Comment (jturel) [inline app/lib/actions/katello/content_view_version/export.rb:5]: Can you add a test for this? Instead of adding it in the test file where you removed the old tests, please create a new test file (the old one is huge) ->  test/actions/katello/content_view_version/export_test.rb

**Review by jturel — CHANGES_REQUESTED** [review]
Minor stuff :)

File: engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/content-view-details.controller.js (hunk 3 above)
Comment (parthaa) [reply engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/content-view-details.controller.js]: I propose completely renaming `::Actions::Pulp3::Orchestration::ContentViewVersion::Export` to ` Actions::Katello::ContentViewVersion::Export` because pulp3 is the only way to export anyway. It makes much more sense in the top level Katello actions directory . We don't need to worry about renaming sub actions, just the top level orchestrate.

Comment (jturel) [reply engines/bastion_katello/app/assets/javascripts/bastion_katello/content-views/details/content-view-details.controller.js]: I'm good with that

**Comment by chris1984** [comment]
@jturel @parthaa how does this look?

**Comment by chris1984** [comment]
[test katello]

**Review by jturel — APPROVED** [review]
ACK!

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]

**Comment by chris1984** [comment]
[test katello]
